bot/exts/utilities/twemoji.py

```diff
@@ -0,0 +1,148 @@
+        return embed
+
+    @staticmethod
+    def emoji(codepoint: str) -> str:
+        """
+        Returns the emoji corresponding to a given `codepoint`.
+
+        The return value is an emoji character, such as "🍂". The `codepoint`
+        argument can be of any format, since it will be trimmed automatically.
+        """
+        if code := Twemoji.trim_code(codepoint):
+            return chr(int(code, 16))
+        return ""
+
+    @staticmethod
+    def codepoint(emoji: str) -> str:
+        """
+        Returns the codepoint, in a trimmed format, of a single emoji.
+
+        `emoji` should be an emoji character, such as "🐍" and "🥰", and
+        not a codepoint like "1f1f8". When working with combined emojis,
+        such as "🇸🇪" and "👨‍👩‍👦", send the component emojis through the method
+        one at a time.
+        """
+        return hex(ord(emoji))[2:]
+
+    @staticmethod
+    def trim_code(codepoint: str | None) -> str:
+        """
+        Returns the meaningful information from the given `codepoint`.
+
+        If no codepoint is found, `None` is returned.
+
+        Example usages:
+        >>> trim_code("U+1f1f8")
+        "1f1f8"
+        >>> trim_code("\u0001f1f8")
+        "1f1f8"
+        >>> trim_code("1f466")
+        "1f466"
+        """
+        if not codepoint:
+            return None
+        if code := CODE.search(codepoint):
+            return code.group()
+
+    def codepoint_from_input(self, raw_emoji: tuple[str, ...]) -> str:
+        """
+        Returns the codepoint corresponding to the passed tuple, separated by "-".
+
+        The return format matches the format used in URLs for Twemoji source files.
+
+        Example usages:
+        >>> codepoint_from_input(["🐍"])
+        "1f40d"
+        >>> codepoint_from_input(["1f1f8", "1f1ea"])
+        "1f1f8-1f1ea"
+        >>> codepoint_from_input(["👨‍👧‍👦"])
+        "1f468-200d-1f467-200d-1f466"
+        """
+        raw_emoji = [emoji.lower() for emoji in raw_emoji]
+        if is_emoji(raw_emoji[0]):
+            emojis = (self.codepoint(emoji) for emoji in raw_emoji[0])
+            return "-".join(emojis)
+
+        raw_first = self.trim_code(raw_emoji[0])
+        if is_emoji("".join(self.emoji(self.trim_code(code)) for code in raw_emoji)):
+            return "-".join(self.trim_code(code) for code in raw_emoji)
+        if is_emoji(self.emoji(raw_first)):
+            return raw_first
+
+        raise ValueError("No codepoint could be obtained from the given input")
+
+    @commands.command(aliases=("tw",))
+    async def twemoji(self, ctx: commands.Context, *raw_emoji: str) -> None:
+        """Sends a preview of a given Twemoji, specified by codepoint or emoji."""
+        if len(raw_emoji) == 0:
+            await invoke_help_command(ctx)
+            return
+        try:
+            codepoint = self.codepoint_from_input(raw_emoji)
+        except (ValueError, AttributeError):
```

Comment:
When do you get an AttributeError? It may be worth to it to attempt to fix what's causing it (unless it's supposed to raise AttributeError but I am failing to see where that would be).

Reply:
I don't quite remember, but I'll investigate. It might've been a past thing.